### ISSUE TITLE
unable to decode Qr code or barcode via QZXingLive on Linux 

### ISSUE THREAD

**Issue by Batiana226** [ftylitak/qzxing#210]
Hello to all, 
I am developing a barcode reader with your library, I want to say that you have done a great job. In my project, the decoding part via an image works perfectly, so I decided to upgrade it with a live decoding. To start, I wanted to test the QZxinglive example on QT, but I can't detect any barcode. My operating system is ubuntu 18 with QT 5.15 and as camera a webcam. I would like to know if someone could help me to solve this problem?

I went through all the solved problems and noticed that it was handled on a branch with the name 86-Fix-QZXingFilter-linux, but even when I go to that branch and run it, I still have the same problem. 

I would like to solve this; any help would be appreciated. 

Thanks

**Comment by ftylitak** [comment]
Hello @Batiana226 , thank you for your kind words.

QZXingLive on master branch should be working out-of-the-box for Ubuntu. For the moment I run it on Ubuntu 21.04 with Qt 5.15. 

To narrow down your issue: 
- Does the camera view starts normally? 
- Do you get any logs in the console output?
- Does it start working if in main.qml, at QZXingFilter element, you change the `tryHarder: false` to `tryHarder: true`?

**Comment by Batiana226** [comment]
the camera view starts, I get data on the output console and the tryharder is false

**Comment by ftylitak** [comment]
In most cases, this means that it has to do with the image decoding and the image format. It can be fixed. 

I would suggest to add the following line: 

```
qDebug() << "format:" << videoFrame.pixelFormat;
```

in line 278: 

https://github.com/ftylitak/qzxing/blob/2fd4dd60c04a29c6d1271fdd9ae25378b8f61ec8/src/QZXingFilter.cpp#L274-L280

to print the image format. It would help a lot if you could send this information.

**Comment by Batiana226** [comment]
In my console i have this format: Format_YUYV

**Comment by ftylitak** [comment]
Greetings, today I have setup a machine with Ubuntu 18.04 and Qt 5.15.2. QZXingLive worked as expected without any need of fixing normally decoding Qr Codes. 

I would recommend to download a clean copy of the QZXing code and retry. Keep in mind that this example, by default has enabled the decoding of `Qr Code`, `EAN 13` and `Code 39`. Should you need any other barcode decoding, add the appropriate macro in the main.qml file : 

https://github.com/ftylitak/qzxing/blob/2fd4dd60c04a29c6d1271fdd9ae25378b8f61ec8/examples/QZXingLive/main.qml#L99 

If this still does not work, it would be helpful to enable frame saving to local disk form further examination. To do this, uncomment the following lines
https://github.com/ftylitak/qzxing/blob/2fd4dd60c04a29c6d1271fdd9ae25378b8f61ec8/src/QZXingFilter.cpp#L383-L387

Start the application, place a Qr Code in front of the camera and send here a representative image from the saved frames to see if there is something wrong.

**Comment by Batiana226** [comment]
Hello, the first suggestion did not work and the second one gives an image like this one. Do you think we can fix this?
![test_75](https://user-images.githubusercontent.com/72884041/132684220-1ec18aa6-ddcd-406b-98eb-018ff6ad259b.png)

**Comment by ftylitak** [comment]
Yes it can be fixed though since I an not yet able to reproduce it, it will take some time. It is related to the parsing of the frame data into a QImage. The responsible code is:

https://github.com/ftylitak/qzxing/blob/2fd4dd60c04a29c6d1271fdd9ae25378b8f61ec8/src/QZXingFilter.cpp#L341-L360

**Comment by ftylitak** [comment]
Sorry for the long delay, though I am a bit puzzled because the frames report that their format is YUYV though the above process can not decode it properly. 

Could you provide the following information for your camera hopping that it would solve a part of the puzzle? 

```bash
sudo apt install v4l-utils

v4l2-ctl v4l2-ctl --list-formats-ext
```

**Comment by majuca** [comment]
HI,

As @Batiana226, I obtain the same result on Windows, the image is not well converted with Qt. 5.15.8. 

Image format: Format_YUYV

Best regards
